src/lib/mina_ledger/test/test_mina_ledger.ml

```
@@ -0,0 +1,197 @@
+                   ~message:
+                     "Account is inserted in stable db backed root, but when \
+                      reloading with unstable DB backed root it's gone "
+               in
+               let expected = LocMap.find_exn loc_with_accounts loc in
+               assert (
+                 Account.equal expected actual_account
+                 || failwith "Inserted account didn't match actual account" ) ) ;
+        Deferred.unit )
+
+  let test_root_moving ~random () =
+    let open L.Root in
+    (* STEP 1: create a root, fill with some accounts, get the locs and closing *)
+    Mina_stdlib_unix.File_system.with_temp_dir "root_moving" ~f:(fun cwd ->
+        (* NOTE: The converting ledger would be created as siblings of the
+           primary ledger, hence we need a sub dir for `with_temp_dir` to clean
+           up the garbages correctly after the test*)
+        (* STEP 1: create a root, fill with some accounts, get the locs and
+           closing *)
+        let backing_type =
+          Quickcheck.Generator.of_list [ Config.Stable_db; Converting_db ]
+          |> Quickcheck.Generator.generate ~size:quickcheck_size ~random
+        in
+        let config =
+          Config.with_directory ~directory_name:(cwd ^/ "ledger") ~backing_type
+        in
+        let root = L.Root.create ~logger ~config ~depth:ledger_depth () in
+        let loc_with_accounts =
+          populate_with_random_accounts ~num:200 ~root ~random
+        in
+        L.Root.close root ;
+        let config_moved =
+          Config.with_directory ~directory_name:(cwd ^/ "ledger_moved")
+            ~backing_type
+        in
+        Config.move_backing_exn ~src:config ~dst:config_moved ;
+        assert (
+          (not (Config.exists_backing config))
+          && Config.exists_backing config_moved
+          || failwith "Config is not moved" ) ;
+        let root_moved =
+          L.Root.create ~logger ~config:config_moved ~depth:ledger_depth ()
```

For the `Converting_db` case, we could consider checking that the two components of the root were kept in sync by moving the root (or checkpointing it, for the other test). There's an unexposed `dbs_synced` method in `converting_merkle_tree`, for instance.

Or, maybe it would be better to add something like a `?fresh:bool` parameter to `With_database.create` and `Root.create` to suppress the automatic migration that `With_database.create` currently does, so we could make sure that the `Root.create` invocation here didn't have to delete and remigrate the converting database.

---

I guess I can add a assert_synced param to converting_merkle_tree, let it fail if it's not synced.